gunrock/applications/color.hxx

```diff
@@ -20,37 +20,33 @@
 namespace gunrock {
 namespace color {
 
-template <typename meta_t>
+template <typename graph_t>
 struct param_t {
   // No parameters for this algorithm
 };
 
-template <typename meta_t>
+template <typename graph_t>
 struct result_t {
-  using vertex_t = typename meta_t::vertex_type;
+  using vertex_t = typename graph_t::vertex_type;
 
   vertex_t* colors;
   result_t(vertex_t* colors_) : colors(colors_) {}
 };
 
-template <typename graph_t,
-          typename meta_t,
-          typename param_t,
-          typename result_t>
-struct problem_t : gunrock::problem_t<graph_t, meta_t, param_t, result_t> {
-  // Use Base class constructor -- does this work? does it handle copy
-  // constructor?
-  using gunrock::problem_t<graph_t, meta_t, param_t, result_t>::problem_t;
-
-  using vertex_t = typename meta_t::vertex_type;
-  using edge_t = typename meta_t::edge_type;
-  using weight_t = typename meta_t::weight_type;
+template <typename graph_t, typename param_type, typename result_type>
+struct problem_t : gunrock::problem_t<graph_t, param_type, result_type> {
+  using gunrock::problem_t<graph_t, param_type, result_type>::problem_t;
+    
```

Comment:
Everything below here is just changes to get `color` to work w/ new `graph_t` API